llama-2-7b/code/web_example_text_completion.py

```diff
@@ -1,68 +1,67 @@
 from flask import Flask, request, jsonify
 from llama import Llama
-# from llama.tokenizer import Tokenizer
+import os
 
 app = Flask(__name__)
 
-class SlidingWindow: 
-    def __init__(self, max_seq_len, max_gen_len): 
+class SlidingWindow:
+    def __init__(self, max_seq_len):
         self.max_seq_len = max_seq_len
-        self.max_gen_len = max_gen_len
         self.token_history = []
 
-    def append(self, new_prompt_tokens): 
+    def append(self, max_gen_len, new_prompt_tokens):
         available_tokens = self.max_seq_len - len(new_prompt_tokens)
-        if available_tokens < 0: 
+        if available_tokens < 0:
             print("User input exceeds the maximum token length")
-        
+
         # Account for tokens required for model output
-        available_tokens -= self.max_gen_len
+        available_tokens -= max_gen_len
 
         if available_tokens > len(self.token_history):
             self.token_history.extend(new_prompt_tokens)
             return self.token_history
-                
+
         # If the total tokens exceed the allowed limit, remove the earliest tokens
         global generator
-        print(generator.tokenizer.decode(self.token_history[-available_tokens:]), "context")
         self.token_history = self.token_history[-available_tokens:] + new_prompt_tokens
-        assert len(self.token_history) == available_tokens
+        assert len(self.token_history) == self.max_seq_len - max_gen_len
```

Comment:
Why is the code change here?

Reply:
I had added the original assert earlier but it is actually incorrect. I had to look more at llama source code.
In the llama model max_seq_len is the maximum sequence length for the input **and output** of the model (https://github.com/facebookresearch/llama/blob/main/llama/generation.py#L122) and (https://www.storagereview.com/review/meta-llama-and-alpacas-loose-in-the-lab-running-large-language-models-locally). For example if we have max_seq_len of 64 and we provide 61 input tokens, llama will only generate 3 output tokens. So our context window needs to occupy the whole sequence space minus space required for output.